pipeline/splicing/calcVarPriors.py

```diff
@@ -139,22 +168,314 @@ def getVarType(variant):
+        if varGenPos < txnStart or varGenPos > txnEnd:
+            return True
+    else:
+        txnStart = int(transcriptData["txEnd"])
+        txnEnd = int(transcriptData["txStart"])
+        if varGenPos > txnStart or varGenPos < txnEnd:
+            return True
+
+    return False
+
+
+def varInUTR(variant):
+    '''
+    Given a variant, if variant is inside transcript boundaries, 
+    determines if variant is in 3' or 5' UTR of transcript
+    '''
+    varGenPos = int(variant["Pos"])
+    varTranscript = variant["Reference_Sequence"]
+    transcriptData = fetch_gene_coordinates(varTranscript)
+    varStrand = getVarStrand(variant)
+    varOutBounds = varOutsideBoundaries(variant)
+    if varOutBounds == False:
+        if varStrand == "+":
+            tsnStart = int(transcriptData["cdsStart"])
+            tsnEnd = int(transcriptData["cdsEnd"])
+            if varGenPos < tsnStart:
+                return True
+            elif varGenPos > tsnEnd:
+                return True
+        else:
+            tsnStart = int(transcriptData["cdsEnd"])
+            tsnEnd = int(transcriptData["cdsStart"])
+            if varGenPos > tsnStart:
+                return True
+            elif varGenPos < tsnEnd:
+                return True
+        return False
+                
+
+def getExonBoundaries(variant):
+    '''
+    Given a variant, returns the exon boundaries for the variant's transcript in a dictionary with format:
+    key = exon number, value = dictionary with exon start and exon end for specific exon
+    Uses function implemented in calcMaxEntScanMeanStd to get data for variant's transcript
+    '''
+    varTranscript = variant["Reference_Sequence"]
+    transcriptData = fetch_gene_coordinates(str(varTranscript))
+
+    # parse exon starts and exon ends
+    transcriptData["exonStarts"] = re.sub(",(\s)*$", "", transcriptData["exonStarts"])
+    transcriptData["exonEnds"] = re.sub(",(\s)*$", "", transcriptData["exonEnds"])
+    if transcriptData["strand"] == "+":
+        exonStarts = transcriptData["exonStarts"].split(",")
+        exonEnds = transcriptData["exonEnds"].split(",")
+    else:
+        # transcript is on '-' strand
+        exonStarts = list(reversed(transcriptData["exonEnds"].split(",")))
+        exonEnds = list(reversed(transcriptData["exonStarts"].split(",")))
+
+    varExons = {}
+    varExonCount = transcriptData["exonCount"]
+    exonCount = 1
+    while exonCount <= varExonCount:
+        exonStart = int(exonStarts[exonCount - 1])
+        exonEnd = int(exonEnds[exonCount - 1])
+        if varTranscript == "NM_007294.3":
+            if exonCount >= 4:
+                # because transcript NM_007294.3 does not include an exon 4, goes from exon 3 to exon 5
+                exonName = "exon" + str(exonCount + 1)
+            else:
+                exonName = "exon" + str(exonCount)
+        else:
+            exonName = "exon" + str(exonCount)
+        varExons[exonName] = {"exonStart": exonStart,
```

Comment:
It does have to be run on each iteration because it is adding the exon start and exon stop for each exon every time it goes through the loop. Unless I'm missing an easier way to do this...

Reply:
ah yea, nm